### PR TITLE
assignment_2 - Refactoring

### PR DESCRIPTION
**Pull request by carolinafm216**

### CODE REVIEW

**Comment by margarida-borbinha** [comment]
Your git workflow wasn't triggered to the refactoring branch
Your code isn't flake8 compliant
You could disable less pylint messages

**Review by margarida-borbinha — CHANGES_REQUESTED** [review]

File: assignments/life_expectancy/__init__.py [hunk 1]
```diff
@@ -1,9 +1,8 @@
 from pathlib import Path
-from life_expectancy.cleaning import clean_data
+from life_expectancy.cleaning import clean_data, file_path
 
 LIFE_EXPECTANCY_FILE_PATH=Path('eu_life_expectancy_raw.tsv')
-#LIFE_EXPECTANCY_FILE_PATH=r"C:\Users\cfmarreiros\Documents\GitHub\FAAST\assignments\life_expectancy\data\eu_life_expectancy_raw.tsv"
-SAVE_FILE_PATH=r"C:\Users\cfmarreiros\Documents\GitHub\FAAST\assignments\life_expectancy\data"
+
```
Comment (MrCordeiro) [inline assignments/life_expectancy/__init__.py:6]: **suggestion**: move this to `main.py`

File: assignments/life_expectancy/cleaning.py [hunk 2]
```diff
@@ -27,34 +23,41 @@ def unpivot(df):
         '1976 ', '1975 ', '1974 ', '1973 ', '1972 ', '1971 ', '1970 ', '1969 ',
         '1968 ', '1967 ', '1966 ', '1965 ', '1964 ', '1963 ', '1962 ', '1961 ',
         '1960 '], value_name="value")
+    #split columns by comma in order to have 4 columns instead of 1
     df[['unit', 'sex', 'age', 'geo\time']] = df[r"unit,sex,age,geo\time"].str.split(",", expand = True)
+    #drop the column thant we split before
     df=df.drop([r'unit,sex,age,geo\time'], axis=1)
+    #rename variables
     df=df.rename(columns={"variable": "year", "geo\time": "region"})
     df = df[['unit', 'sex', 'age', 'region', 'year', 'value']]
```
Comment (MrCordeiro) [inline assignments/life_expectancy/cleaning.py:32]: df=df
df = df

File: assignments/life_expectancy/cleaning.py [hunk 3]
```diff
@@ -27,34 +23,41 @@ def unpivot(df):
...
     return df
 
 def date_format(df):
```
Comment (MrCordeiro) [inline assignments/life_expectancy/cleaning.py:35]:
```suggestion
def _convert_date_format(df: pd.DataFrame) -> pd.DataFrame:
```

**Review by MrCordeiro — CHANGES_REQUESTED** [review]

File: assignments/life_expectancy/tests/test_cleaning.py [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
 import pandas as pd
 
-from life_expectancy.cleaning import clean_data
+from life_expectancy.cleaning import clean_data, save_data,load_data, file_path
 from . import OUTPUT_DIR
 
 
 def test_clean_data(pt_life_expectancy_expected):
     """Run the `clean_data` function and compare the output to the expected output"""
     #clean_data()
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/tests/test_cleaning.py:10]: Feito

File: assignments/life_expectancy/cleaning.py [hunk 5]
```diff
@@ -27,34 +23,41 @@ def unpivot(df):
...
-def filter_region(df,country):
+def filter_region(df, country):
     df = df[df.region == country]
     return df
 
 def save_data(df):
     df.to_csv(pathlib.Path(__file__).parent/"data"/ "pt_life_expectancy.csv",index=False)
 
+def clean_data(df,country):
+    return filter_region(date_format(unpivot(df)), country)
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/cleaning.py:55]: feito

File: assignments/life_expectancy/cleaning.py (hunk 3 above)
Comment (carolinafm216) [reply assignments/life_expectancy/cleaning.py:35]: feito

File: assignments/life_expectancy/tests/test_cleaning.py [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
-    clean_data(OUTPUT_DIR / "eu_life_expectancy_raw.tsv",'PT')
+    save_data(clean_data(load_data(file_path), 'PT'))
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/tests/test_cleaning.py:11]: feito

File: assignments/life_expectancy/cleaning.py [hunk 7]
```diff
@@ -27,34 +23,41 @@ def unpivot(df):
...
 if __name__ == "__main__":  # pragma: no cover
     parser = argparse.ArgumentParser()
     parser.add_argument("country")
     args = parser.parse_args()
 
-    read_file_name = "eu_life_expectancy_raw.tsv"
-    file_path = pathlib.Path(__file__).parent/"data"/ read_file_name
-
-    clean_data(file_path,args.country)
+    save_data(clean_data(load_data(file_path), args.country))
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/cleaning.py:63]: feito

File: assignments/life_expectancy/cleaning.py [hunk 8]
```diff
@@ -2,21 +2,17 @@
...
+file_path = pathlib.Path(__file__).parent/"data"/ read_file_name
 
-def clean_data (path:pathlib.Path, country):
-    save_data(filter_region(date_format(unpivot(load_file(path))), country))
-
-
-def load_file(file_name):
-    with open(file_name, encoding="utf8") as file:
+def load_data(path:pathlib.Path):
+    with open(path, encoding="utf8") as file:
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/cleaning.py:9]: feito

File: assignments/life_expectancy/cleaning.py [hunk 9]
```diff
@@ -2,21 +2,17 @@
...
         read_file=pd.read_csv(file,sep='\t')
         df = pd.DataFrame(read_file)
     return df
 
 def unpivot(df):
+    #keep unit, sex, age and geo time as columns and unpivot years
     df = df.melt(id_vars=[r"unit,sex,age,geo\time"],
                         value_vars=['2021 ','2020 ', '2019 ', '2018 ', '2017 ',
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/cleaning.py:17]: feito

File: assignments/life_expectancy/cleaning.py (hunk 2 above)
Comment (carolinafm216) [reply assignments/life_expectancy/cleaning.py:32]: feito (black)

**Review by margarida-borbinha — APPROVED** [review]

**Pull request merged**